### PR TITLE
New tag mappings and refactor of tag names

### PR DESCRIPTION
**Pull request by Earthcomputer**
This PR refactors the old (poor quality) tag mappings and also adds names for the new tag stuff.

### CODE REVIEW

File: mappings/net/minecraft/tag/TagGroup.mapping [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CLASS net/minecraft/class_5414 net/minecraft/tag/TagGroup
+	COMMENT Contains the set of tags all of the same type
```
Comment (liach) [inline mappings/net/minecraft/tag/TagGroup.mapping]:
```suggestion
	COMMENT Contains the set of tags all of the same type.
```

**Review by liach — APPROVED** [review]
A significant improvement over previous names.

**Review by lenrik1589 — APPROVED** [review]
Seems ok to me.

File: mappings/net/minecraft/tag/SetTag.mapping [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+		ARG 2 type
 	METHOD method_29898 empty ()Lnet/minecraft/class_5394;
-	METHOD method_29900 (Ljava/util/Set;)Lnet/minecraft/class_5394;
-		ARG 0 set
+	METHOD method_29899 getCommonType (Ljava/lang/Class;Ljava/lang/Class;)Ljava/lang/Class;
+		ARG 0 first
+		ARG 1 second
+	METHOD method_29900 of (Ljava/util/Set;)Lnet/minecraft/class_5394;
+		ARG 0 values
+	METHOD method_29901 getCommonType (Ljava/util/Set;)Ljava/lang/Class;
```
Comment (i509VCB) [inline mappings/net/minecraft/tag/SetTag.mapping:14]: @liach take notes for #1516

**Review by i509VCB — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Looks good

**Review by shartte — APPROVED** [review]
LGTM

**Pull request merged**